homeassistant/components/media_player/denonavr.py

```diff
@@ -41,11 +41,33 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Denon platform."""
     import denonavr
 
-    receiver = denonavr.DenonAVR(config.get(CONF_HOST), config.get(CONF_NAME))
-
-    add_devices([DenonDevice(receiver)])
-    _LOGGER.info("Denon receiver at host %s initialized",
-                 config.get(CONF_HOST))
+    # Initialize name and host
+    host = None
+    name = None
+
+    # Start assignment of host and name
+    # 1. option: manual setting
+    if config.get(CONF_HOST) is not None:
+        host = config.get(CONF_HOST)
+        name = config.get(CONF_NAME)
+    # 2. option: discovery using netdisco
+    elif discovery_info is not None:
```

Comment:
This is not going to work like this.

Discovery is going to run from time to time and when finding a Denon receiver will call `setup_platform` without a config.

So best to keep track of Denon receivers is by storing a set of hosts in the dict `hass.data`. This is a dictionary that is alive for the duration of Home Assistant and in which you can store info.

```python
KEY_DENON_CACHE = 'denonavr_hosts'

cache = hass.data.get(KEY_DENON_CACHE)

if cache is None:
    cache = hass.data[KEY_DENON_CACHE] = set()

# Find hosts

for host in hosts:
    if host in cache:
        continue

    # Actually setup host
```